src/context/user-context.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { createContext, useContext, useState } from 'react'
+
+export type UserProps = {
+  name: string
+  email: string
+}
+
+export type UserContextProps = {
+  user: UserProps
+  setUser: (values: object) => void
+}
+
+const UserContext = createContext({} as UserContextProps)
```

Comment:
Existe alguma necessidade específica para a criação desse contexto? Ele basicamente define um estado e compartilha com os `children` o getter e setter do mesmo. Talvez faça mais sentido um contexto de autenticação, um `AuthContext` que encapsula as regras e implementações relacionadas a sign in, sign out, refresh token e a persistência dos mesmos no `async storage` ou `indexeddb`. É sempre bom evitar a criação de contextos descenessários ou, quando criar, refinar o máximo possível, levando em consideração que sempre que uma tela ou componente depende de um contexto e esse contexto muda, teríamos rerenders desnecessários, independente de utilizar as técnicas de memoização disponíveis pelo React

Reply:
Eu apaguei algumas coisas com o intuito de deixa-lo mais limpo e mais fácil de fazer o setup